internal/integrationtest/profiles/profiles_test.go

```diff
@@ -0,0 +1,86 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package profiles_test
+
+import (
+	"testing"
+
+	"github.com/arduino/arduino-cli/internal/integrationtest"
+	"github.com/arduino/go-paths-helper"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCompileWithProfiles(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	// Init the environment explicitly
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+
+	// copy sketch_with_profile into the working directory
+	sketchName := "sketch_with_profile"
+	p, err := paths.Getwd()
+	require.NoError(t, err)
+	require.NotNil(t, p)
+	testSketch := p.Parent().Join("testdata", sketchName)
+	sketchPath := cli.SketchbookDir().Join(sketchName)
+	err = testSketch.CopyDirTo(sketchPath)
+	require.NoError(t, err)
```

Comment:
@cmaglie maybe this could be made part of the `testsuite` library? Just like we did for python version: the https://github.com/arduino/arduino-cli/blob/aba6a789058e0572b4ed5b88558f70b62d6e1e17/test/conftest.py#L227-L236

Reply:
yep

Reply:
@MatteoPologruto actually you are able to do it yourself... A good place to put this function would be https://github.com/arduino/arduino-cli/blob/db6897aeaebc4d486db8a74d9ad0b6f7e5665276/internal/integrationtest/arduino-cli.go

Reply:
Sure, I'll handle it in another PR.

Reply:
Tests updated using the CopySketch function, added [here](https://github.com/arduino/arduino-cli/pull/1879).